custom/icds_reports/management/commands/citus_migration.py

```diff
@@ -0,0 +1,254 @@
+    res = connection.execute("""
+        SELECT c.relname AS child, p.relname as parent
+        FROM pg_inherits JOIN pg_class AS c ON (inhrelid=c.oid)
+        JOIN pg_class as p ON (inhparent=p.oid);
+    """)
+
+    mapping = defaultdict(list)
+    for row in res:
+        mapping[row.parent].append(row.child)
+    return mapping
+
+
+def get_all_tables(connection):
+    res = connection.execute("select tablename from pg_tables where schemaname = 'public'")
+    return {row.tablename for row in res}
+
+
+def table_exists(connection, table_name):
+    res = connection.execute("select 1 from pg_tables where tablename = %s", table_name)
+    return bool(list(res))
+
+
+def get_dump_command(table, rename_to=None):
+    dump = 'pg_dump -t {table} $SOURCE_DB $DUMP_OPTS'
+    if rename_to:
+        dump += ' | sed "s/{table}/{rename_to}/g"'
+    dump += ' | psql $CITUS_CONNECT'
+    return dump.format(table=table, rename_to=rename_to)
+
+
+def get_dump_pre_commands(source_config, target_config):
+    return inspect.cleandoc("""
+    #!/bin/bash
+
+    SOURCE_DB='{source_db}'
+    CITUS_CONNECT="-h {target_host} -U {target_user} {target_db}
+    DUMP_OPTS="--data-only --no-acl"
+    """.format(
+        source_db=source_config['NAME'],
+        target_host=target_config['HOST'],
+        target_user=target_config['USER'],
+        target_db=target_config['NAME']
+    ))
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'action',
+            choices=['update-schema', 'generate-dump-script'],
+        )
+        parser.add_argument(
+            '--source-db-alias', default='icds-ucr',
+            help='Django alias for source database'
+        )
+        parser.add_argument(
+            '--target-db-alias', default='icds-ucr-citus',
+            help='Django alias for target database'
+        )
+        parser.add_argument(
+            '--citus-user',
+            help='PG user to connect to CitusDB as. Defaults to the user in localsettings.',
+        )
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            help='Only output logging.',
+        )
+
+    def handle(self, action, source_db_alias, target_db_alias, **options):
+        self.dry_run = options['dry_run']
+
+        with connection_manager.get_engine(source_db_alias).begin() as conn:
+            self.parent_child_mapping = get_parent_child_mapping(conn)
+            self.child_parent_mapping = {
+                child: parent
+                for parent, children in self.parent_child_mapping.items()
+                for child in children
+            }
+
+        if action == 'update-schema':
+            self.update_schema(source_db_alias, target_db_alias)
+
+        elif action == 'generate-dump-script':
+            self.generate_dump_script(options, source_db_alias, target_db_alias)
+
+    def generate_dump_script(self, options, source_db_alias, target_db_alias):
```

Comment:
Is there a way to stop/resume the script (either in case of failure or when needed for anything)? If we track all commands in a stub SQL table, we could do it in a more failsafe way (the way we did for backfilling? It probably doesn't matter, if we expect the script to finish with a day or two.

Reply:
How about something like this:
```
#!/bin/bash

start_at=$1

echo "Starting from $start_at"

if [ -z "$start_at" -o "$start_at" = "1" ]; then
    start_at=""
    echo '1'
fi

if [ -z "$start_at" -o "$start_at" = "2" ]; then
    start_at=""
    echo '2'
fi

if [ -z "$start_at" -o "$start_at" = "3" ]; then
    start_at=""
    echo '3'
fi
```